<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qSlicerExternalBeamPlanningModule</class>
 <widget class="qSlicerWidget" name="qSlicerExternalBeamPlanningModule">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>543</width>
    <height>779</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>RT Plan</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_3">
   <item row="2" column="0">
    <widget class="ctkCollapsibleButton" name="CTKCollapsibleButton_ExternalBeamPlanning">
     <property name="text">
      <string>RT Plan</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <property name="margin">
       <number>4</number>
      </property>
      <property name="spacing">
       <number>4</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Reference Volume:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="qMRMLNodeComboBox" name="MRMLNodeComboBox_RtPlan">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="toolTip">
         <string/>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLRTPlanNode</string>
         </stringlist>
        </property>
        <property name="addEnabled">
         <bool>true</bool>
        </property>
        <property name="removeEnabled">
         <bool>true</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
        <property name="selectNodeUponCreation">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_ReferenceVolume">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>RT Plan:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="MRMLNodeComboBox_ReferenceVolume">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLScalarVolumeNode</string>
         </stringlist>
        </property>
        <property name="addEnabled">
         <bool>false</bool>
        </property>
        <property name="removeEnabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QPushButton" name="pushButton_AddBeam">
          <property name="text">
           <string>Add Beam</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_RemoveBeam">
          <property name="text">
           <string>Remove Beam</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QTableWidget" name="tableWidget_Beams">
        <property name="columnCount">
         <number>4</number>
        </property>
        <attribute name="horizontalHeaderVisible">
         <bool>true</bool>
        </attribute>
        <attribute name="horizontalHeaderDefaultSectionSize">
         <number>70</number>
        </attribute>
        <attribute name="horizontalHeaderHighlightSections">
         <bool>false</bool>
        </attribute>
        <attribute name="horizontalHeaderMinimumSectionSize">
         <number>10</number>
        </attribute>
        <column>
         <property name="text">
          <string>Vis</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Name</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Type</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Radiation</string>
         </property>
        </column>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <property name="sizeConstraint">
      <enum>QLayout::SetFixedSize</enum>
     </property>
     <item>
      <widget class="QLabel" name="label_13">
       <property name="text">
        <string>Parameter Set:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="qMRMLNodeComboBox" name="MRMLNodeComboBox_ParameterSet">
       <property name="nodeTypes">
        <stringlist>
         <string>vtkMRMLExternalBeamPlanningNode</string>
        </stringlist>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="3" column="0">
    <widget class="ctkCollapsibleButton" name="CollapsibleButton_BeamParameters">
     <property name="text">
      <string>Beam Parameters</string>
     </property>
     <property name="contentsFrameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Beam Name:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLineEdit" name="lineEdit_BeamName"/>
      </item>
      <item row="3" column="0" colspan="3">
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QPushButton" name="pushButton_BeamPrescription">
          <property name="text">
           <string>Prescription</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_BeamGeometry">
          <property name="text">
           <string>Geometry</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_BeamModel">
          <property name="text">
           <string>Beam Model</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Radiation Type:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QComboBox" name="comboBox_RadiationType">
        <item>
         <property name="text">
          <string>Proton</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Photon</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Electron</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="4" column="0" colspan="3">
       <widget class="QStackedWidget" name="stackedWidget">
        <property name="currentIndex">
         <number>1</number>
        </property>
        <widget class="QWidget" name="photonPrescriptionPage">
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <property name="margin">
           <number>0</number>
          </property>
          <item>
           <layout class="QFormLayout" name="formLayout_2">
            <property name="fieldGrowthPolicy">
             <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
            </property>
            <item row="1" column="0">
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>Beam Type:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QComboBox" name="comboBox_BeamType">
              <item>
               <property name="text">
                <string>Static</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Dynamic</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_12">
              <property name="text">
               <string>Rx Dose (Gy):</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QLineEdit" name="lineEdit_RxDose"/>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_NominalEnergy">
              <property name="text">
               <string>Nominal Energy (kVp):</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QComboBox" name="comboBox_NominalEnergy">
              <item>
               <property name="text">
                <string>80</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>120</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="label_BeamOnTime">
              <property name="text">
               <string>Beam On Time (S):</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QLineEdit" name="lineEdit_BeamOnTime">
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <widget class="QLabel" name="label_NominalmA">
              <property name="text">
               <string>Nominal mA:</string>
              </property>
             </widget>
            </item>
            <item row="5" column="1">
             <widget class="QLineEdit" name="lineEdit_NominalmA"/>
            </item>
            <item row="6" column="0">
             <widget class="QLabel" name="label_CollimatorType">
              <property name="text">
               <string>Collimator:</string>
              </property>
             </widget>
            </item>
            <item row="6" column="1">
             <widget class="QComboBox" name="comboBox_CollimatorType">
              <item>
               <property name="text">
                <string>Square 0.5x0.5</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Square 1.0x1.0</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="protonGeometryPage">
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <property name="margin">
           <number>0</number>
          </property>
          <item>
           <layout class="QFormLayout" name="formLayout">
            <property name="fieldGrowthPolicy">
             <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="label_15">
              <property name="text">
               <string>Target volume</string>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="label_14">
              <property name="text">
               <string>Smearing</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QLineEdit" name="lineEdit_Smearing">
              <property name="text">
               <string>5.0</string>
              </property>
              <property name="readOnly">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>Proximal margin</string>
              </property>
             </widget>
            </item>
            <item row="5" column="1">
             <widget class="QLineEdit" name="lineEdit_ProximalMargin">
              <property name="text">
               <string>3.0</string>
              </property>
              <property name="readOnly">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="qMRMLNodeComboBox" name="MRMLNodeComboBox_ProtonTargetContour">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="nodeTypes">
               <stringlist>
                <string>vtkMRMLContourNode</string>
               </stringlist>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label">
              <property name="frameShape">
               <enum>QFrame::NoFrame</enum>
              </property>
              <property name="text">
               <string>Isocenter</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="qMRMLNodeComboBox" name="MRMLNodeComboBox_IsocenterFiducial">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="nodeTypes">
               <stringlist>
                <string>vtkMRMLMarkupsFiducialNode</string>
               </stringlist>
              </property>
              <property name="showChildNodeTypes">
               <bool>true</bool>
              </property>
              <property name="addEnabled">
               <bool>false</bool>
              </property>
              <property name="removeEnabled">
               <bool>false</bool>
              </property>
              <property name="renameEnabled">
               <bool>false</bool>
              </property>
              <property name="selectNodeUponCreation">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Gantry angle:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="ctkSliderWidget" name="SliderWidget_GantryAngle">
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="singleStep">
               <double>1.000000000000000</double>
              </property>
              <property name="pageStep">
               <double>10.000000000000000</double>
              </property>
              <property name="maximum">
               <double>359.000000000000000</double>
              </property>
              <property name="tickInterval">
               <double>18.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="ctkSliderWidget" name="SliderWidget_CouchAngle">
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>359.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Couch angle</string>
              </property>
             </widget>
            </item>
            <item row="6" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Distal margin</string>
              </property>
             </widget>
            </item>
            <item row="6" column="1">
             <widget class="QLineEdit" name="lineEdit_DistalMargin">
              <property name="text">
               <string>8.0</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="beamModelPage">
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <property name="margin">
           <number>0</number>
          </property>
          <item>
           <layout class="QFormLayout" name="formLayout_3">
            <property name="fieldGrowthPolicy">
             <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="label_16">
              <property name="text">
               <string>Source distance</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="lineEdit_SourceDistance">
              <property name="text">
               <string>2000</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Energy resolution</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QLineEdit" name="lineEdit_EnergyResolution">
              <property name="text">
               <string>1</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_11">
              <property name="text">
               <string>Energy spread</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QLineEdit" name="lineEdit_EnergySpread">
              <property name="text">
               <string>1</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_17">
              <property name="text">
               <string>Source size</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLineEdit" name="lineEdit_SourceSize">
              <property name="text">
               <string>20</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="0,0,0,0">
     <property name="sizeConstraint">
      <enum>QLayout::SetFixedSize</enum>
     </property>
     <item>
      <widget class="QLabel" name="label_CalculateDoseStatus">
       <property name="text">
        <string>Status information here</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_CalculateWED">
       <property name="text">
        <string>Calculate WED</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_CalculateDose">
       <property name="text">
        <string>Calculate Dose</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>qSlicerWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkSliderWidget</class>
   <extends>QWidget</extends>
   <header>ctkSliderWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>qSlicerExternalBeamPlanningModule</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>MRMLNodeComboBox_ReferenceVolume</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>343</x>
     <y>3</y>
    </hint>
    <hint type="destinationlabel">
     <x>448</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerExternalBeamPlanningModule</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>MRMLNodeComboBox_ParameterSet</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>363</x>
     <y>2</y>
    </hint>
    <hint type="destinationlabel">
     <x>363</x>
     <y>10</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerExternalBeamPlanningModule</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>MRMLNodeComboBox_RtPlan</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>71</x>
     <y>735</y>
    </hint>
    <hint type="destinationlabel">
     <x>202</x>
     <y>96</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerExternalBeamPlanningModule</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>MRMLNodeComboBox_IsocenterFiducial</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>160</x>
     <y>614</y>
    </hint>
    <hint type="destinationlabel">
     <x>317</x>
     <y>548</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerExternalBeamPlanningModule</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>MRMLNodeComboBox_ProtonTargetContour</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>271</x>
     <y>389</y>
    </hint>
    <hint type="destinationlabel">
     <x>317</x>
     <y>663</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
